src/v/features/enterprise_features.h

```diff
@@ -40,13 +40,12 @@ std::ostream& operator<<(std::ostream&, license_required_feature);
  */
 class enterprise_feature_report {
     using vtype = absl::flat_hash_set<license_required_feature>;
-    using range = boost::iterator_range<vtype::const_iterator>;
 
 public:
     void set(license_required_feature feat, bool enabled);
     bool test(license_required_feature);
-    range enabled() const { return _enabled; }
-    range disabled() const { return _disabled; }
+    const vtype& enabled() const { return _enabled; }
+    const vtype& disabled() const { return _disabled; }
```

Comment:
out of curiosity, did this have any unwanted side effects?

Reply:
Not that I can think of. Both are only used for iterating over or printing the elements. There's no side effect for iteration and the side effect for printing is intentional. (Because `absl::flat_hash_set<>` doesn't have a default printer, it's more difficult to make the mistake that this PR fixes.)